Tests/FoundationTests/NSAttributedStringExtensionsTests.swift

```diff
@@ -70,6 +70,27 @@ final class NSAttributedStringExtensionsTests: XCTestCase {
     }
 
     // MARK: - Methods
+    func testApplying() {
+        #if canImport(AppKit) || canImport(UIKit)
+        let string = NSAttributedString(string: "Applying")
+        var out = string.applying(attributes: [:])
+        var attributes = out.attributes(at: 0, effectiveRange: nil)
+        XCTAssertTrue(attributes.isEmpty)
+
+        out = string.applying(attributes: [
+            .strikethroughStyle: NSNumber(value: NSUnderlineStyle.single.rawValue as Int),
+            .foregroundColor: Color.red
+        ])
+        attributes = out.attributes(at: 0, effectiveRange: nil)
+        let filteredAttributes = attributes.filter { (key, value) -> Bool in
```

Comment:
Do you even need to filter them, since you know that there were already no attributes earlier?

Reply:
The first test passes an empty `Dictionary` to the `applying(attributes:)` method. The second test case passes `Dictionary` with attribute values.

Reply:
What happens if you call `XCTAssertEqual(attributes.count, 2)`? It should pass, right? Filtering them will return the same dictionary.

Reply:
I retain the filter closure just to make sure that the `applying(attributes:)` returns the attributes that were passed and does not modify/add different attribute. But yeah, `XCTAssertEqual(attributes.count, 2)` will pass.

Reply:
Then check the values in `attributes`.
```Swift
XCTAssertEqual(attributes[.foregroundColor], UIColor.red)
XCTAssertEqual(attributes[.strikethroughStyle], NSUnderlineStyle.single)
```